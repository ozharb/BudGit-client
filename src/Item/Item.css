
.Item { 
    padding: 10px;
   border-radius: 20px;
   border-style: solid;
   border-color: white;  
   filter:drop-shadow(5px 10px black);
   margin: 20px;
   background: linear-gradient(black, rgb(27, 27, 27) );
}
.Item-Header{
justify-content: space-between;
display: flex;
  cursor: pointer;  
}

button.Item__expand-item-button{
    font-size: 30px;
}
.MainList__edit-item-button{
    background-color:Transparent;
    border-radius: 20px;
    border-width: 4px;
    border-style: solid;
    border-color: white;
    color: white;
    display: block;
    font-size: 12px;
    font-weight: bold;
    height: 5px;
    text-align: center;
    text-decoration: none;
    width: 7px;
    padding-right: 30px;
    padding-bottom: 20px;
    padding-left: 15px;
    margin-left: 30px;
}

.MainList__edit-item-button:hover{
   
    border-color: #FF0038;
    color: #FF0038;
    filter:drop-shadow(0px 0px 8px #FF0038);
  } 
.MainList__edit-item-button:focus{
     border-color: #FF0038;
     color: #FF0038;
     filter:drop-shadow(0px 0px 8px #FF0038);
  } 

h3.Item__name {
font-size: 40px;
}
  .Item__date{
      font-size: 15px;
      color: gray;
  }
  .Item > button{
cursor: pointer;
  }
  .item__delete{
cursor: pointer;
  }
  .item__delete:hover{
color: #D3002E;
filter:drop-shadow(0px 0px 8px #D3002E);
      }
.item__delete:focus{
color: #D3002E;
filter:drop-shadow(0px 0px 8px #D3002E);
}
.item__calc{
cursor: pointer;
}
.item__calc:hover{
color: #3CACD5;
filter:drop-shadow(0px 0px 8px #3CACD5);
}
.item__calc:focus{
color: #3CACD5;
filter:drop-shadow(0px 0px 8px #3CACD5);
}
  .put-back-item{
    color: #3CACD5;
    filter:drop-shadow(0px 0px 8px #149BCC);
  }
.item-cost-qty{
    font-size: 30px;
    display: flex;
    justify-content: space-evenly;
}
.item-remove-delete{
font-style: 30px;
}
  @media (max-width: 700px) {
    button.Item__expand-item-button{
        font-size: 20px;
    }
    h3.Item__name {
        font-size: 20px;
        }
    .Item__name{
    font-size: 15px;
    }
    .Item{
        margin:2px;
    }
    .Item-Header{
    font-size: 10px;
    }
.Item__date-made{
font-size: 12px;
}
.item-remove-delete{
    margin-top: 10px;
}
.item-cost-qty{
font-size: 15px;
}
    }