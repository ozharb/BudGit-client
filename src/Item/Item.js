import React, {Component} from 'react'
import { Link } from 'react-router-dom'
import { parseISO, toDate, format } from 'date-fns-tz'
// import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
// import {faSkullCrossbones} from '@fortawesome/free-solid-svg-icons';
import PropTypes from 'prop-types';
import './Item.css'
import ApiContext from '../ApiContext'
import config from '../config'
import AppButton from '../AppButton/AppButton'
import ExpandButton from '../ExpandButton/ExpandButton'
import { getItemsForList, findItem } from '../app-helpers'


class Item extends Component {
  
  state = {
    expand: false,
};

static contextType = ApiContext;
handleChangeCalc = e => {
    e.preventDefault()
    const itemId = parseInt(this.props.id)
    const { items=[] } = this.context
    const item = findItem(items, itemId)
    const newItem = {...item, calc: !item.calc}
   
 fetch(config.API_ENDPOINT + `/items/${itemId}`, {
      method: 'PATCH',
      body: JSON.stringify(newItem),
      headers: {
        'content-type': 'application/json',
      },
    })
      .then(res => {
        if (!res.ok)
          return res.json().then(error => Promise.reject(error))
        })
        .then(() => {
    
         
            this.context.handleUpdate(newItem)

    
        })
        .catch(error => {
          console.error({ error })
        })
  };

 handleItemExpand = e => {
   this.setState({
     expand: !this.state.expand
   })
 } 
handleClickDelete = e => {
  e.preventDefault()
  const itemId = parseInt(this.props.id)


  fetch(`${config.API_ENDPOINT}/items/${itemId}`, {
    method: 'DELETE',
    headers: {
      'content-type': 'application/json'
    },
  })
    .then(res => {
      if (!res.ok)
        return res.json().then(e => Promise.reject(e))
      
    })
    .then(() => {

     
      this.context.deleteItem(itemId)

    })
    .catch(error => {
      console.error({ error })
    })
}


render(){
    const itemId = parseInt(this.props.id)
    const { items=[] } = this.context
    const item = findItem(items, itemId)
    console.log('details:', item.content)
    const expandItemButtonText = this.state.expand
? `﹀` 
:  `〉`
const itemDetails= this.state.expand
? <div className='Item__content'>
{item.content.split(/\n \r|\n/).map((para, i) =>
  <p key={i}>{para}</p>
)}
 <AppButton
  tag={Link}
  to={`/edit/${itemId}`}
        //   component={AddItem}
          className='MainList__edit-item-button'
        >
          {/* <FontAwesomeIcon icon='plus' /> */}
          Edit
        </AppButton>
</div>
:  null
    const calcButton = item.calc
    ? 'Remove'
     : 'Put Back'

  const { name, id, price, quantity, date_made, calc } = this.props
  return (
    <div className='Item'>
      <header className='Item-Header'>
            <ExpandButton
          type='button'
          onClick={this.handleItemExpand}
          className='Item__expand-item-button'>
          {expandItemButtonText}
        </ExpandButton>
      <h2 className='Item__name'>
        {/* <Link to={`/item/${id}`}> */}
          {name}
        {' | '}
          cost: ${price}
        {' | '}
          qty: {quantity}
        {/* </Link> */}
      </h2>
    
  
        </header>
  {itemDetails}
  <br />
      <button className = 'item_calc'
      type='radio' name = 'calc' 
      onClick={this.handleChangeCalc} 
      
      value = {calc}> 
      
      {calcButton}
      
      </button>

      <button className='item__delete' 
      type='button'
      onClick={this.handleClickDelete}
      >
         
        {/* <FontAwesomeIcon icon={faSkullCrossbones}  /> */}
        {' '}
        Delete
      </button>
      <div className='Item__date'>
        <div className='Item__date_made'>
          Added:
          {' '}
          <span className='date_made'>
            {
format(toDate(date_made), 'EEEE MM/dd/yyyy h:mm aaaaa\'m\' z', { timeZone : 'America/Los_Angeles' })}
          </span>
        </div>
      </div>
    </div>
  )
}
}
Item.defaultProps = {
  name:"",
  id: "",
}

Item.propTypes = {
  props: PropTypes.shape({
    id: PropTypes.isRequired,
    item_name: PropTypes.string.isRequired,
  }),

}
export default Item