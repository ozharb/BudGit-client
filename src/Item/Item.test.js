import React from 'react';
import Item from './Item';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom'

it('renders the UI as expected', () => {
  const tree = renderer
    
    .create(
    <BrowserRouter>
    <Item props={{item: {calc: true, item_name:'none'}}} date_made={0}/>
     </BrowserRouter>
    )

    
    .toJSON();
 
  expect(tree).toMatchSnapshot();  
  });