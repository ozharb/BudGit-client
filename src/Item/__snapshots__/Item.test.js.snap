// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="Item"
>
  <header
    className="Item-Header"
  >
    <div
      className="item-name-expand"
    >
      <button
        className="Item__expand-item-button"
        onClick={[Function]}
        type="button"
      >
        <i
          className="fas fa-chevron-right"
        />
      </button>
    </div>
    <h3
      className="Item__name"
      onClick={[Function]}
    />
    <div
      role="alert"
    />
    <div
      className="item-remove-delete"
    >
      <div
        className="remove-button"
      >
        <button
          className="item__calc"
          name="calc"
          onClick={[Function]}
          type="radio"
        >
          <i
            className="fas fa-circle"
          />
        </button>
      </div>
      <div
        className="delete-button"
      >
        <button
          className="item__delete"
          onClick={[Function]}
          type="button"
        >
          <i
            className="fas fa-times"
          />
        </button>
      </div>
    </div>
  </header>
  <div
    className="item-cost-qty"
  >
    <div
      className="item-cost-display"
    >
         cost: $
       
    </div>
    <div
      className="item-qty-display"
    >
       qty: 
    </div>
  </div>
  <br />
  <div
    className="Item__date"
  >
    <div
      className="Item__date-made"
    >
      Added:
       
      <span
        className="date-made"
      >
        Wednesday 12/31/1969
      </span>
    </div>
  </div>
</div>
`;
